src/common/components/select-repository-list/index.js

```diff
@@ -202,7 +190,16 @@ function mapStateToProps(state) {
     snaps,
     repositories,
     repositoriesStatus,
-    selectRepositoriesForm
+    selectRepositoriesForm,
+    isAddingSnaps: isAddingSnaps(state),
+    isRepoSelected: (repo) => hasRepository(selectRepositoriesForm.selectedRepos, repo),
```

Comment:
Selectors (as far as I currently understand them) allow to compute some derived data based on current state.

These functions here (`isRepoSelected` and `isRepoEnabled`) also compute derived data based on a state, but only for given repository.

Yes, they are kind of helper functions and could live outside of the class and be passed the data instead of getting it from state here.
But for me they felt similar enough to selectors to include them as props here.

As mentioned by @earnubs they probably would be simplified (or not needed at all) once snap store is refactored. So maybe we can revisit it then?